### ISSUE TITLE
Change `Capability` to trait `AsRef<str>`

### ISSUE THREAD

**Issue by kgv** [aatxe/irc#25]
```rust
pub trait ServerExt<'a, T, U>: Server<'a, T, U> {
...
    /// Sends an IRCv3 capabilities request for the specified extensions.
    fn send_cap_req(&self, extensions: &[Capability]) -> Result<()> where Self: Sized {
...
```

Propose: Change `Capability` to trait `AsRef<str>`.

Because some servers use custom capabilities ([for example](https://github.com/justintv/Twitch-API/blob/master/chat/capabilities.md))

**Comment by aatxe** [comment]
Would it suffice to add a Custom capability? My thinking is that it's fitting since some (maybe most?) capabilities will actually require some sort of tangible support beyond just sending a request for it. So, it might then make sense to still keep the enumeration around. Your alternative is also perfectly reasonable. It's not dissimilar to some of the concerns with the current Command design and potential redesign options in #16. Do you think it'd be preferred ergonomically to just use strings for requesting capabilities?

**Comment by kgv** [comment]
I think that keeping the enumeration and adding Custom variant will be more ergonomically.

**Comment by aatxe** [comment]
This was implemented in a5306f83918a4a4b17826844e5fac7d4fe6a9498.
